packages/@aws-cdk/aws-appsync/lib/data-source.ts

```diff
@@ -81,13 +103,41 @@ export interface ExtendedDataSourceProps {
 }
 
 /**
- * Abstract AppSync datasource implementation. Do not use directly but use subclasses for concrete datasources
+ * An Interface for Data Sources
  */
-export abstract class BaseDataSource extends Construct {
+export interface IDataSource extends IConstruct {
+  /**
+   * the name of the data source
+   */
+  readonly name: string;
+  /**
+   * the arn of the data source
+   */
+  readonly arn: string;
+  /**
+   * the underlying CFN data source resource
+   */
+  readonly ds: CfnDataSource;
```

Comment:
we should pick a better name than `ds`
I also don't think the `Cfn` type should be expected here.

we should use the properties that users would provide when creating a data source (which would not expose Cfn properties) for supporting imports

Reply:
this existed before and i didnt know if it would be a good idea to remove it.. if we make DataSource a Resource we could use the `node.defaultChild` escape hatch?